### PR TITLE
Remove deprecated throw declarations

### PR DESCRIPTION
**Pull request by Flamefire**
Fixed those in libPMacc
This is a showstopper for e.g. CUDA 7!
@psychocoderHPC @n01r 

**Environment**
CUDA 7.0.27
gcc (Ubuntu 4.8.4-2ubuntu1~14.04) 4.8.4

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
Can you please explain why `throw (std::runtime_error)` is needed?
What is the error message without `throw ...)`

**Comment by Flamefire** [comment]
Source for deprecation: http://www.cplusplus.com/doc/tutorial/exceptions/

**Comment by Flamefire** [comment]
Removed all exception specifications now

**Comment by ax3l** [comment]
did you use a c++11 host compiler? we don't support c++11 yet.

**Comment by ax3l** [comment]
P.S.: issue 1k! wuhu!

**Comment by psychocoderHPC** [comment]
Wait its only issue 1k because he did a request to master before, see #999. IMO it was purely intended :-)

**Comment by Flamefire** [comment]
Compiler:
Cuda compilation tools, release 7.0, V7.0.27
gcc (Ubuntu 4.8.4-2ubuntu1~14.04) 4.8.4

And yes it is a C++11 compatible compiler and in one case C++11 is used (in user code). This doesn't break anything in old compilers and should be removed for future C++11 compatibility.

Also the missing header IS a bug.

**Comment by ax3l** [comment]
ok.
pls split bugs from c++11 features in independent pulls.

we were actually planning to change to c++11 in one step, but if @psychocoderHPC does not mind you can provide the first features/refactorings (clearly stated as those in a independent pull pls) :)

**Comment by Flamefire** [comment]
AFAIK Benjamin already removed some of the throw stuff and these are probably bugs introduced by not removing all of them.
Do you really consider this a C++11 "feature"? Its more a compatibility thing...

**Comment by BenjaminW3** [comment]
This is not only a C++11 compatibility thing.
It is also an optimization because exception specifications are checked at runtime and not at compile time as I stated in #913.

I am for removing them but please add a new pull request for the added header bug.

**Comment by psychocoderHPC** [comment]
The problem is not only related to C++11.

Description of the problem:

- in some  interfaces we define `virtual ~CLASSNAME()`
- we inherit from such interfaces without defining the destructor  of the master class `virtual`

The mixed cases results in wrong destructor signatures. 
Two solutions are possible:
- add `virtual` to destructor of some classes ([deprecated](http://www.cplusplus.com/doc/tutorial/exceptions/))
- remove `throw ...` from the the method signature (current pull request)

**Comment by Flamefire** [comment]
The problem is that virtual method overriding others with throw(...) must use the same exception specification. So you can't have a base method without throw and a derived one with it or the other way round. This is a bug, not a C++11 feature!
2 PRs on this already merged: https://github.com/ComputationalRadiationPhysics/picongpu/pull/913 https://github.com/ComputationalRadiationPhysics/picongpu/pull/918

**Comment by BenjaminW3** [comment]
Have you removed all occurences of dynamic exception specifications in all files?

**Comment by Flamefire** [comment]
grep -rE "throw *\(" . did not show anything else.

**Comment by BenjaminW3** [comment]
I have marked it as a bug because the current version has incompatible dynamic exception specifications between ```TaskGetCurrentSizeFromDevice``` and its base classes because I missed them in my commits.
A pedantic compiler will issue an error on this.
By removing all the exception specifications in the whole project we will be future proof without any drawbacks.

**Comment by psychocoderHPC** [comment]
Can we please split this pull request in `PMacc` and `tools` changes. This helps the change log write (@ax3l :-)

**Comment by ax3l** [comment]
all right, thanks for catching up!

File: src/libPMacc/include/eventSystem/tasks/TaskGetCurrentSizeFromDevice.hpp [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /**
- * Copyright 2013-2015 Felix Schmitt, Rene Widera, Benjamin Worpitz
+ * Copyright 2013-2015 Felix Schmitt, Rene Widera, Benjamin Worpitz,
+ * Alexander Grund
```
Comment (BenjaminW3) [inline src/libPMacc/include/eventSystem/tasks/TaskGetCurrentSizeFromDevice.hpp]: Could you please indent your name to align with the first name in the first line?

File: src/libPMacc/include/eventSystem/tasks/TaskSetCurrentSizeOnDevice.hpp [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /**
- * Copyright 2013-2015 Felix Schmitt, Rene Widera, Benjamin Worpitz
+ * Copyright 2013-2015 Felix Schmitt, Rene Widera, Benjamin Worpitz,
+ * Alexander Grund
```
Comment (BenjaminW3) [inline src/libPMacc/include/eventSystem/tasks/TaskSetCurrentSizeOnDevice.hpp]: alignment

File: src/libPMacc/include/memory/buffers/DeviceBufferIntern.hpp [hunk 3]
```diff
@@ -1,5 +1,6 @@
 /**
- * Copyright 2013-2015 Axel Huebl, Heiko Burau, Rene Widera, Benjamin Worpitz
+ * Copyright 2013-2015 Axel Huebl, Heiko Burau, Rene Widera, Benjamin Worpitz,
+ * Alexander Grund
```
Comment (BenjaminW3) [inline src/libPMacc/include/memory/buffers/DeviceBufferIntern.hpp]: alignment

File: src/libPMacc/include/memory/buffers/HostBufferIntern.hpp [hunk 4]
```diff
@@ -1,5 +1,6 @@
 /**
- * Copyright 2013-2015 Rene Widera, Benjamin Worpitz
+ * Copyright 2013-2015 Rene Widera, Benjamin Worpitz,
+ * Alexander Grund
```
Comment (BenjaminW3) [inline src/libPMacc/include/memory/buffers/HostBufferIntern.hpp]: alignement

File: src/libPMacc/include/memory/buffers/MappedBufferIntern.hpp [hunk 5]
```diff
@@ -1,5 +1,6 @@
 /**
- * Copyright 2014-2015 Rene Widera, Axel Huebl, Benjamin Worpitz
+ * Copyright 2014-2015 Rene Widera, Axel Huebl, Benjamin Worpitz,
+ * Alexander Grund
```
Comment (BenjaminW3) [inline src/libPMacc/include/memory/buffers/MappedBufferIntern.hpp]: alignment

**Pull request merged**